env0/resource_environment.go

```diff
@@ -1058,7 +1058,7 @@ func getConfigurationVariableFromSchema(variable map[string]interface{}) client.
 	return configurationVariable
 }
 
-func getEnvironmentByName(name interface{}, meta interface{}, excludeArchived bool) (client.Environment, diag.Diagnostics) {
+func getEnvironmentByName(meta interface{}, name string, projectId string, excludeArchived bool) (client.Environment, diag.Diagnostics) {
```

Comment:
added a new field and changed the order of the variables.
In case more variables are added, I'll convert it to a "struct filter".

Reply:
d